eurec4a_environment/plot/profile.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import xarray as xr
+import matplotlib.dates as mdates
+import seaborn as sns
+import typhon
+
+sns.set(
+    context="notebook",
+    style="white",
+    palette="deep",
+    font="sans-serif",
+    font_scale=2,
+    color_codes=True,
+    rc=None,
+)
+
+
+def profile_plot_2D(
+    ds,
+    variable,
+    ax=None,
+    y="alt",
+    x="launch_time",
+    cbar_label=None,
+    height_labels=None,
+    height_labels_color="green",
+    **kwargs
+):
+
+    """Plot a 2D (default time-height) plot and optionally another height level as scatter points
+
+    **kwargs: Additional keyword arguments passed to
+            :func:`matplotlib.pyplot.contourf`.
+    """
+
+    # ensure we have x (default launch time) as a coordinate
+    time = x
+    org_dim = None
+    try:
+        ds.isel(**{time: 0})
+    except ValueError:
+        org_dim = ds[time].dims[0]
+        ds = ds.swap_dims({org_dim: time})
+
+    # Plot
+    fig, ax = plt.subplots(figsize=(20, 10))
+
+    var = ds[variable]
+
+    var.plot.contourf(
```

Comment:
For cases with irregular time intervals or "discontinuities", using pcolormesh instead of contourf might be better (e.g. for airborne platforms). Maybe we can provide an option for the user to decide between pcolormesh or contourf. The option can be simply interpolate_between_profiles=True or False, so that they don't have to bother with the function name.

Reply:
Okay!

Reply:
Sounds good to me too ⭐